blockinfo.go

```diff
@@ -0,0 +1,116 @@
+/*
+Package weave defines all common interfaces to weave
+together the various subpackages, as well as
+implementations of some of the simpler components
+(when interfaces would be too much overhead).
+
+We pass a BlockInfo struct with all framework-defined
+information down the Decorator/Handler stack.
+For custom info that is only to be consumed within a particular
+module, or timeouts, etc, you can make use of
+context.Context. Please do not store info in there that is required
+for other code to work, rather optional context to enhance functionality
+*/
+package weave
+
+import (
+	"regexp"
+	"time"
+
+	"github.com/iov-one/weave/errors"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/libs/log"
+)
+
+var (
+	// DefaultLogger is used for all context that have not
+	// set anything themselves
+	DefaultLogger = log.NewNopLogger()
+
+	// IsValidChainID is the RegExp to ensure valid chain IDs
+	IsValidChainID = regexp.MustCompile(`^[a-zA-Z0-9_\-]{6,20}$`).MatchString
+)
+
+type BlockInfo struct {
```

Comment:
If `BlockInfo` is intended as not only block information container but as a general purpose request scope context, than maybe different name would be better. `RequestScope` or `RequestContext` :see_no_evil: 

This implementation has all fields related to block information, except for the `logger`. Logger does not seem to fit this structure at all.

Reply:
`RequestScope` is a nice idea.

Logger should be passed down somewhere, and sticking it in Context seemed like the uglier hack. Other solutions? I think we rarely if ever actually log anything, so maybe just remove it :see_no_evil:

Reply:
I agree with the logger not really fitting here. To pass it with the context was an anti pattern already (see https://dave.cheney.net/tag/logging). Instead let's set them explicitly with the constructors where needed.

Reply:
Agreed about logger. 

Don't like the `Scope` naming, I don't think it's a name for a struct at all - gives me the wrong idea.
`RequestScope` sounds more like a constant to describe which scope we are in to execute some tasks that are intended for this scope. It also seems to me that we can't really operate with the term `Request` on blockchain, at least not with the current struct content as it gives the wrong idea. 

I would much rather prefer something like `ChainStatus` or keep `BlockInfo`. 

I can't come up with a better name at the moment, but please don't go for `Scope` :)

Reply:
Good point with the block chain terminology.

How about `State` (used as `weave.State`)? This structure represents the current state of the application.